packages/compiler/core/checker.ts

```diff
@@ -2254,7 +2270,14 @@ export function createChecker(program: Program): Checker {
     const modelSymId = getNodeSymId(model);
     pendingResolutions.add(modelSymId);
     let isType;
-    if (isExpr.kind === SyntaxKind.ArrayExpression) {
+    if (isExpr.kind == SyntaxKind.ModelExpression) {
```

Comment:
```suggestion
    if (isExpr.kind === SyntaxKind.ModelExpression) {
```

Reply:
should we add eslint rule for that? surprised it is not coming as part of the recommended ones